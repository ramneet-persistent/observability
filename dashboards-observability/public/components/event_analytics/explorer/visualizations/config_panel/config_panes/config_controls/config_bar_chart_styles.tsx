/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useMemo, useCallback, Fragment } from 'react';
import { EuiAccordion, EuiSpacer } from '@elastic/eui';
import { get } from 'lodash';
import { ButtonGroupItem } from './config_button_group';
import { IConfigPanelOptionSection } from '../../../../../../../../common/types/explorer';
import {
  SLIDER_MIN_VALUE,
  SLIDER_MAX_VALUE,
  SLIDER_DEFAULT_STEP,
} from '../../../../../../../../common/constants/shared';
import { ConfigChartOptionsEnum } from '../../../../../../../../common/constants/explorer';

export const ConfigBarChartStyles = ({
  visualizations,
  schemas,
  vizState,
  handleConfigChange,
  sectionName,
  sectionId = 'chartStyles',
}: any) => {
  const { data } = visualizations;
  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;

  const handleConfigurationChange = useCallback(
    (stateFieldName) => {
      return (changes) => {
        handleConfigChange({
          ...vizState,
          [stateFieldName]: changes,
        });
      };
    },
    [handleConfigChange, vizState]
  );

  /* To update the schema options based on current style mode selection */
  const currentSchemas = useMemo(() => {
    if (vizState?.orientation === 'h') {
      return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'labelAngle');
    }
    return schemas;
  }, [vizState]);

  const dimensions = useMemo(
    () =>
      currentSchemas.map((schema: IConfigPanelOptionSection, index: number) => {
        let params = {
          title: schema.name,
          vizState,
          ...schema.props,
        };
        const DimensionComponent = schema.component || ButtonGroupItem;

        const createDimensionComponent = (dimProps) => (
          <Fragment key={`viz-series-${index}`}>
            <DimensionComponent {...dimProps} />
            <EuiSpacer size="s" />
          </Fragment>
        );
        switch (schema.eleType) {
          case ConfigChartOptionsEnum.buttons:
            params = {
              ...params,
              legend: schema.name,
              groupOptions: schema?.props?.options.map((btn: { name: string }) => ({
                ...btn,
                label: btn.name,
              })),
              idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
              handleButtonChange: handleConfigurationChange(schema.mapTo),
            };
            return createDimensionComponent(params);
          case ConfigChartOptionsEnum.input:
            params = {
              title: schema.name,
              currentValue: vizState[schema.mapTo] || '',
              numValue: vizState[schema.mapTo] || '',
              handleInputChange: handleConfigurationChange(schema.mapTo),
              vizState,
              ...schema.props,
            };
            return createDimensionComponent(params);

          case ConfigChartOptionsEnum.slider:
            params = {
              ...params,
              minRange:
                typeof get(schema, 'props.min') === undefined
                  ? SLIDER_MIN_VALUE
                  : get(schema, 'props.min'),
              maxRange:
                typeof get(schema, 'props.max') === undefined
                  ? SLIDER_MAX_VALUE
                  : get(schema, 'props.max'),
              step: schema?.props?.step || SLIDER_DEFAULT_STEP,
              currentRange: vizState[schema.mapTo] || schema?.defaultState,
              ticks: schema?.props?.ticks,
              showTicks: schema?.props?.showTicks || false,
              handleSliderChange: handleConfigurationChange(schema.mapTo),
            };
            return createDimensionComponent(params);

          default:
            params = {
              ...params,
              paddingTitle: schema.name,
              advancedTitle: 'advancedTitle',
              dropdownList:
                schema?.options?.map((option) => ({ ...option })) ||
                fields.map((item) => ({ ...item })),
              onSelectChange: handleConfigurationChange(schema.mapTo),
              isSingleSelection: schema.isSingleSelection,
              selectedAxis: vizState[schema.mapTo] || schema.defaultState,
            };
            return createDimensionComponent(params);
        }
      }),
    [currentSchemas, vizState, handleConfigurationChange]
  );

  return (
    <EuiAccordion
      initialIsOpen
      id={`configPanel__${sectionId}`}
      buttonContent={sectionName}
      paddingSize="s"
    >
      {dimensions}
    </EuiAccordion>
  );
};
