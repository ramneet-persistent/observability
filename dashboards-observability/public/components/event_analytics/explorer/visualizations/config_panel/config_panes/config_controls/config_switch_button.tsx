/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiSwitch, EuiFormRow, EuiSpacer } from '@elastic/eui';

interface SwitchButtonProps {
  currentValue: boolean;
  title: string;
  onToggle: (value: boolean) => void;
}
export const SwitchButton = ({
  currentValue,
  onToggle,
  title,
}: SwitchButtonProps) => {
  return (
    <>
      {/* <EuiFormRow display="columnCompressedSwitch"  label={title}> */}
      <EuiSpacer />
      <EuiSwitch
        label={title}
        checked={currentValue}
        onChange={(e) => onToggle(e.target.checked)}
      />
      {/* </EuiFormRow> */}
    </>
  );
};
