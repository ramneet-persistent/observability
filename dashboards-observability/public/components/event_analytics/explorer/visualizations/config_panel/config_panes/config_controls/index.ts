/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { ConfigPanelOptions } from './config_panel_options';
export { ConfigValueOptions } from './config_value_options';
export { ConfigChartOptions } from './config_chart_options';
export { ConfigDataLinks } from './config_data_links';
export { ConfigThresholds } from './config_thresholds';
export { ConfigText } from './config_text';
export { ConfigGaugeValueOptions } from './config_gauge_options';
export { ConfigInputField } from "./config_input_field";
export { ConfigGridColumns } from "./config_grid_columns"
export { ConfigColorTheme } from './config_color_theme';

