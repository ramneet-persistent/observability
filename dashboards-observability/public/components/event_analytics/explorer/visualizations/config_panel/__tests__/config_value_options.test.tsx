/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { ConfigValueOptions } from '../config_panes/config_controls/config_value_options';
import { TEST_VISUALIZATIONS_DATA } from '../../../../../../../test/event_analytics_constants';

describe('ConfigValueOptions component', () => {
  configure({ adapter: new Adapter() });

  it('Renders ConfigValueOptions with empty data', async () => {
    const wrapper = mount(
      <ConfigValueOptions
        visualizations={TEST_VISUALIZATIONS_DATA}
        schemas={[]}
        vizState={[]}
        handleConfigChange={jest.fn()}
        sectionName={'Test name'}
        sectionId={'valueOptions'}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Renders ConfigValueOptions with data', async () => {
    const wrapper = mount(
      <ConfigValueOptions
        visualizations={TEST_VISUALIZATIONS_DATA}
        schemas={[]}
        vizState={[]}
        handleConfigChange={jest.fn()}
        sectionName={'Test name'}
        sectionId={'valueOptions'}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
