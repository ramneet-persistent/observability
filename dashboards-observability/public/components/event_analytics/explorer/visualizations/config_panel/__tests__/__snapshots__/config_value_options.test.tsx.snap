// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigValueOptions component Renders ConfigValueOptions with data 1`] = `
<ConfigValueOptions
  handleConfigChange={[MockFunction]}
  schemas={Array []}
  sectionId="valueOptions"
  sectionName="Test name"
  visualizations={
    Object {
      "data": Object {
        "appData": Object {
          "fromApp": false,
        },
        "defaultAxes": Object {},
        "indexFields": Object {
          "availableFields": Array [
            Object {
              "name": "agent",
              "type": "string",
            },
            Object {
              "name": "bytes",
              "type": "long",
            },
            Object {
              "name": "clientip",
              "type": "ip",
            },
            Object {
              "name": "event",
              "type": "struct",
            },
            Object {
              "name": "extension",
              "type": "string",
            },
            Object {
              "name": "geo",
              "type": "struct",
            },
            Object {
              "name": "host",
              "type": "string",
            },
            Object {
              "name": "index",
              "type": "string",
            },
            Object {
              "name": "ip",
              "type": "ip",
            },
            Object {
              "name": "machine",
              "type": "struct",
            },
            Object {
              "name": "memory",
              "type": "double",
            },
            Object {
              "name": "message",
              "type": "string",
            },
            Object {
              "name": "phpmemory",
              "type": "long",
            },
            Object {
              "name": "referer",
              "type": "string",
            },
            Object {
              "name": "request",
              "type": "string",
            },
            Object {
              "name": "response",
              "type": "string",
            },
            Object {
              "name": "tags",
              "type": "string",
            },
            Object {
              "name": "timestamp",
              "type": "timestamp",
            },
            Object {
              "name": "url",
              "type": "string",
            },
            Object {
              "name": "utc_time",
              "type": "timestamp",
            },
          ],
          "queriedFields": Array [
            Object {
              "name": "double_per_ip_bytes",
              "type": "long",
            },
            Object {
              "name": "host",
              "type": "text",
            },
            Object {
              "name": "ip_count",
              "type": "integer",
            },
            Object {
              "name": "per_ip_bytes",
              "type": "long",
            },
            Object {
              "name": "resp_code",
              "type": "text",
            },
            Object {
              "name": "sum_bytes",
              "type": "long",
            },
          ],
          "selectedFields": Array [],
          "unselectedFields": Array [],
        },
        "query": Object {},
        "rawVizData": Object {
          "data": Object {
            "count()": Array [
              154,
              1753,
              116,
              468,
              1964,
              219,
            ],
            "tags": Array [
              "error",
              "info",
              "login",
              "security",
              "success",
              "warning",
            ],
          },
          "jsonData": Array [
            Object {
              "count()": 154,
              "tags": "error",
            },
            Object {
              "count()": 1753,
              "tags": "info",
            },
            Object {
              "count()": 116,
              "tags": "login",
            },
            Object {
              "count()": 468,
              "tags": "security",
            },
            Object {
              "count()": 1964,
              "tags": "success",
            },
            Object {
              "count()": 219,
              "tags": "warning",
            },
          ],
          "metadata": Object {
            "fields": Array [
              Object {
                "name": "count()",
                "type": "integer",
              },
              Object {
                "name": "tags",
                "type": "text",
              },
            ],
            "size": 6,
            "status": 200,
          },
        },
        "userConfigs": Object {
          "dataConfig": Object {
            "valueOptions": Object {
              "dimensions": Array [
                Object {
                  "label": "tags",
                  "name": "tags",
                  "type": "text",
                },
              ],
              "metrics": Array [
                Object {
                  "label": "count()",
                  "name": "count()",
                  "side": "left",
                  "type": "integer",
                },
              ],
            },
          },
        },
      },
      "vis": Object {
        "barwidth": 0.97,
        "category": "Visualizations",
        "categoryaxis": "xaxis",
        "component": [Function],
        "editorconfig": Object {
          "panelTabs": Array [
            Object {
              "editor": [Function],
              "id": "data-panel",
              "mapTo": "dataConfig",
              "name": "Style",
              "sections": Array [
                Object {
                  "editor": [Function],
                  "id": "tooltip_options",
                  "mapTo": "tooltipOptions",
                  "name": "Tooltip options",
                  "schemas": Array [
                    Object {
                      "component": null,
                      "mapTo": "tooltipMode",
                      "name": "Tooltip mode",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                          Object {
                            "id": "hidden",
                            "name": "Hidden",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": null,
                      "mapTo": "tooltipText",
                      "name": "Tooltip text",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "all",
                            "name": "All",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "all",
                            "name": "All",
                          },
                          Object {
                            "id": "x",
                            "name": "Dimension",
                          },
                          Object {
                            "id": "y",
                            "name": "Series",
                          },
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "editor": [Function],
                  "id": "legend",
                  "mapTo": "legend",
                  "name": "Legend",
                  "schemas": Array [
                    Object {
                      "component": null,
                      "mapTo": "showLegend",
                      "name": "Show legend",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                          Object {
                            "id": "hidden",
                            "name": "Hidden",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": null,
                      "mapTo": "position",
                      "name": "Position",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "v",
                            "name": "Right",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "v",
                            "name": "Right",
                          },
                          Object {
                            "id": "h",
                            "name": "Bottom",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": [Function],
                      "eleType": "input",
                      "mapTo": "legendSize",
                      "name": "Legend size",
                      "title": "Legend size",
                    },
                  ],
                },
                Object {
                  "editor": [Function],
                  "id": "chart_styles",
                  "mapTo": "chartStyles",
                  "name": "Chart styles",
                  "schemas": Array [
                    Object {
                      "component": [Function],
                      "eleType": "buttons",
                      "mapTo": "mode",
                      "name": "Mode",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "group",
                            "name": "Group",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "group",
                            "name": "Group",
                          },
                          Object {
                            "id": "stack",
                            "name": "Stack",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": [Function],
                      "eleType": "input",
                      "mapTo": "labelSize",
                      "name": "Label size",
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 0,
                      "eleType": "slider",
                      "mapTo": "rotateBarLabels",
                      "name": "Rotate bar labels",
                      "props": Object {
                        "max": 90,
                        "min": -90,
                        "showTicks": true,
                        "ticks": Array [
                          Object {
                            "label": "-90°",
                            "value": -90,
                          },
                          Object {
                            "label": "-45°",
                            "value": -45,
                          },
                          Object {
                            "label": "0°",
                            "value": 0,
                          },
                          Object {
                            "label": "45°",
                            "value": 45,
                          },
                          Object {
                            "label": "90°",
                            "value": 90,
                          },
                        ],
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 0.7,
                      "eleType": "slider",
                      "mapTo": "groupWidth",
                      "name": "Group width",
                      "props": Object {
                        "max": 1,
                        "step": 0.01,
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 0.97,
                      "eleType": "slider",
                      "mapTo": "barWidth",
                      "name": "Bar width",
                      "props": Object {
                        "max": 1,
                        "step": 0.01,
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 2,
                      "eleType": "slider",
                      "mapTo": "lineWidth",
                      "name": "Line width",
                      "props": Object {
                        "max": 10,
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 70,
                      "eleType": "slider",
                      "mapTo": "fillOpacity",
                      "name": "Fill opacity",
                      "props": Object {
                        "max": 100,
                      },
                    },
                  ],
                },
              ],
            },
            Object {
              "editor": [Function],
              "id": "availability-panel",
              "mapTo": "availabilityConfig",
              "name": "Availability",
            },
          ],
        },
        "fillopacity": 70,
        "fulllabel": "Vertical bar",
        "groupwidth": 0.7,
        "icon": [Function],
        "icontype": "visBarVerticalStacked",
        "id": "bar",
        "label": "Vertical bar",
        "labelangle": 0,
        "legendposition": "v",
        "linewidth": 2,
        "mode": "group",
        "name": "bar",
        "orientation": "v",
        "selection": Object {
          "dataLoss": "nothing",
        },
        "seriesaxis": "yaxis",
        "showlegend": "show",
        "type": "bar",
        "visconfig": Object {
          "config": Object {
            "displaylogo": false,
            "responsive": true,
          },
          "isUniColor": false,
          "layout": Object {
            "height": 1180,
            "legend": Object {
              "orientation": "v",
              "traceorder": "normal",
            },
            "margin": Object {
              "b": 30,
              "l": 60,
              "pad": 0,
              "r": 30,
              "t": 50,
            },
            "showlegend": true,
          },
        },
      },
    }
  }
  vizState={Array []}
>
  <EuiAccordion
    arrowDisplay="left"
    buttonContent="Test name"
    id="configPanel__valueOptions"
    initialIsOpen={true}
    isLoading={false}
    isLoadingMessage={false}
    paddingSize="s"
  >
    <div
      className="euiAccordion euiAccordion-isOpen"
    >
      <div
        className="euiAccordion__triggerWrapper"
      >
        <button
          aria-controls="configPanel__valueOptions"
          aria-expanded={true}
          className="euiAccordion__button"
          id="random_html_id"
          onClick={[Function]}
          type="button"
        >
          <span
            className="euiAccordion__iconWrapper"
          >
            <EuiIcon
              className="euiAccordion__icon euiAccordion__icon-isOpen"
              size="m"
              type="arrowRight"
            >
              <EuiIconArrowRight
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiAccordion__icon euiAccordion__icon-isOpen"
                focusable="false"
                role="img"
                style={null}
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiAccordion__icon euiAccordion__icon-isOpen"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.157 13.069l4.611-4.685a.546.546 0 000-.768L5.158 2.93a.552.552 0 010-.771.53.53 0 01.759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 01-.76 0 .552.552 0 010-.771z"
                    fillRule="nonzero"
                  />
                </svg>
              </EuiIconArrowRight>
            </EuiIcon>
          </span>
          <span
            className="euiIEFlexWrapFix"
          >
            Test name
          </span>
        </button>
      </div>
      <div
        aria-labelledby="random_html_id"
        className="euiAccordion__childWrapper"
        id="configPanel__valueOptions"
        role="region"
        tabIndex={-1}
      >
        <EuiResizeObserver
          onResize={[Function]}
        >
          <div>
            <div
              className="euiAccordion__padding--s"
            />
          </div>
        </EuiResizeObserver>
      </div>
    </div>
  </EuiAccordion>
</ConfigValueOptions>
`;

exports[`ConfigValueOptions component Renders ConfigValueOptions with empty data 1`] = `
<ConfigValueOptions
  handleConfigChange={[MockFunction]}
  schemas={Array []}
  sectionId="valueOptions"
  sectionName="Test name"
  visualizations={
    Object {
      "data": Object {
        "appData": Object {
          "fromApp": false,
        },
        "defaultAxes": Object {},
        "indexFields": Object {
          "availableFields": Array [
            Object {
              "name": "agent",
              "type": "string",
            },
            Object {
              "name": "bytes",
              "type": "long",
            },
            Object {
              "name": "clientip",
              "type": "ip",
            },
            Object {
              "name": "event",
              "type": "struct",
            },
            Object {
              "name": "extension",
              "type": "string",
            },
            Object {
              "name": "geo",
              "type": "struct",
            },
            Object {
              "name": "host",
              "type": "string",
            },
            Object {
              "name": "index",
              "type": "string",
            },
            Object {
              "name": "ip",
              "type": "ip",
            },
            Object {
              "name": "machine",
              "type": "struct",
            },
            Object {
              "name": "memory",
              "type": "double",
            },
            Object {
              "name": "message",
              "type": "string",
            },
            Object {
              "name": "phpmemory",
              "type": "long",
            },
            Object {
              "name": "referer",
              "type": "string",
            },
            Object {
              "name": "request",
              "type": "string",
            },
            Object {
              "name": "response",
              "type": "string",
            },
            Object {
              "name": "tags",
              "type": "string",
            },
            Object {
              "name": "timestamp",
              "type": "timestamp",
            },
            Object {
              "name": "url",
              "type": "string",
            },
            Object {
              "name": "utc_time",
              "type": "timestamp",
            },
          ],
          "queriedFields": Array [
            Object {
              "name": "double_per_ip_bytes",
              "type": "long",
            },
            Object {
              "name": "host",
              "type": "text",
            },
            Object {
              "name": "ip_count",
              "type": "integer",
            },
            Object {
              "name": "per_ip_bytes",
              "type": "long",
            },
            Object {
              "name": "resp_code",
              "type": "text",
            },
            Object {
              "name": "sum_bytes",
              "type": "long",
            },
          ],
          "selectedFields": Array [],
          "unselectedFields": Array [],
        },
        "query": Object {},
        "rawVizData": Object {
          "data": Object {
            "count()": Array [
              154,
              1753,
              116,
              468,
              1964,
              219,
            ],
            "tags": Array [
              "error",
              "info",
              "login",
              "security",
              "success",
              "warning",
            ],
          },
          "jsonData": Array [
            Object {
              "count()": 154,
              "tags": "error",
            },
            Object {
              "count()": 1753,
              "tags": "info",
            },
            Object {
              "count()": 116,
              "tags": "login",
            },
            Object {
              "count()": 468,
              "tags": "security",
            },
            Object {
              "count()": 1964,
              "tags": "success",
            },
            Object {
              "count()": 219,
              "tags": "warning",
            },
          ],
          "metadata": Object {
            "fields": Array [
              Object {
                "name": "count()",
                "type": "integer",
              },
              Object {
                "name": "tags",
                "type": "text",
              },
            ],
            "size": 6,
            "status": 200,
          },
        },
        "userConfigs": Object {
          "dataConfig": Object {
            "valueOptions": Object {
              "dimensions": Array [
                Object {
                  "label": "tags",
                  "name": "tags",
                  "type": "text",
                },
              ],
              "metrics": Array [
                Object {
                  "label": "count()",
                  "name": "count()",
                  "side": "left",
                  "type": "integer",
                },
              ],
            },
          },
        },
      },
      "vis": Object {
        "barwidth": 0.97,
        "category": "Visualizations",
        "categoryaxis": "xaxis",
        "component": [Function],
        "editorconfig": Object {
          "panelTabs": Array [
            Object {
              "editor": [Function],
              "id": "data-panel",
              "mapTo": "dataConfig",
              "name": "Style",
              "sections": Array [
                Object {
                  "editor": [Function],
                  "id": "tooltip_options",
                  "mapTo": "tooltipOptions",
                  "name": "Tooltip options",
                  "schemas": Array [
                    Object {
                      "component": null,
                      "mapTo": "tooltipMode",
                      "name": "Tooltip mode",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                          Object {
                            "id": "hidden",
                            "name": "Hidden",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": null,
                      "mapTo": "tooltipText",
                      "name": "Tooltip text",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "all",
                            "name": "All",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "all",
                            "name": "All",
                          },
                          Object {
                            "id": "x",
                            "name": "Dimension",
                          },
                          Object {
                            "id": "y",
                            "name": "Series",
                          },
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "editor": [Function],
                  "id": "legend",
                  "mapTo": "legend",
                  "name": "Legend",
                  "schemas": Array [
                    Object {
                      "component": null,
                      "mapTo": "showLegend",
                      "name": "Show legend",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "show",
                            "name": "Show",
                          },
                          Object {
                            "id": "hidden",
                            "name": "Hidden",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": null,
                      "mapTo": "position",
                      "name": "Position",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "v",
                            "name": "Right",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "v",
                            "name": "Right",
                          },
                          Object {
                            "id": "h",
                            "name": "Bottom",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": [Function],
                      "eleType": "input",
                      "mapTo": "legendSize",
                      "name": "Legend size",
                      "title": "Legend size",
                    },
                  ],
                },
                Object {
                  "editor": [Function],
                  "id": "chart_styles",
                  "mapTo": "chartStyles",
                  "name": "Chart styles",
                  "schemas": Array [
                    Object {
                      "component": [Function],
                      "eleType": "buttons",
                      "mapTo": "mode",
                      "name": "Mode",
                      "props": Object {
                        "defaultSelections": Array [
                          Object {
                            "id": "group",
                            "name": "Group",
                          },
                        ],
                        "options": Array [
                          Object {
                            "id": "group",
                            "name": "Group",
                          },
                          Object {
                            "id": "stack",
                            "name": "Stack",
                          },
                        ],
                      },
                    },
                    Object {
                      "component": [Function],
                      "eleType": "input",
                      "mapTo": "labelSize",
                      "name": "Label size",
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 0,
                      "eleType": "slider",
                      "mapTo": "rotateBarLabels",
                      "name": "Rotate bar labels",
                      "props": Object {
                        "max": 90,
                        "min": -90,
                        "showTicks": true,
                        "ticks": Array [
                          Object {
                            "label": "-90°",
                            "value": -90,
                          },
                          Object {
                            "label": "-45°",
                            "value": -45,
                          },
                          Object {
                            "label": "0°",
                            "value": 0,
                          },
                          Object {
                            "label": "45°",
                            "value": 45,
                          },
                          Object {
                            "label": "90°",
                            "value": 90,
                          },
                        ],
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 0.7,
                      "eleType": "slider",
                      "mapTo": "groupWidth",
                      "name": "Group width",
                      "props": Object {
                        "max": 1,
                        "step": 0.01,
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 0.97,
                      "eleType": "slider",
                      "mapTo": "barWidth",
                      "name": "Bar width",
                      "props": Object {
                        "max": 1,
                        "step": 0.01,
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 2,
                      "eleType": "slider",
                      "mapTo": "lineWidth",
                      "name": "Line width",
                      "props": Object {
                        "max": 10,
                      },
                    },
                    Object {
                      "component": [Function],
                      "defaultState": 70,
                      "eleType": "slider",
                      "mapTo": "fillOpacity",
                      "name": "Fill opacity",
                      "props": Object {
                        "max": 100,
                      },
                    },
                  ],
                },
              ],
            },
            Object {
              "editor": [Function],
              "id": "availability-panel",
              "mapTo": "availabilityConfig",
              "name": "Availability",
            },
          ],
        },
        "fillopacity": 70,
        "fulllabel": "Vertical bar",
        "groupwidth": 0.7,
        "icon": [Function],
        "icontype": "visBarVerticalStacked",
        "id": "bar",
        "label": "Vertical bar",
        "labelangle": 0,
        "legendposition": "v",
        "linewidth": 2,
        "mode": "group",
        "name": "bar",
        "orientation": "v",
        "selection": Object {
          "dataLoss": "nothing",
        },
        "seriesaxis": "yaxis",
        "showlegend": "show",
        "type": "bar",
        "visconfig": Object {
          "config": Object {
            "displaylogo": false,
            "responsive": true,
          },
          "isUniColor": false,
          "layout": Object {
            "height": 1180,
            "legend": Object {
              "orientation": "v",
              "traceorder": "normal",
            },
            "margin": Object {
              "b": 30,
              "l": 60,
              "pad": 0,
              "r": 30,
              "t": 50,
            },
            "showlegend": true,
          },
        },
      },
    }
  }
  vizState={Array []}
>
  <EuiAccordion
    arrowDisplay="left"
    buttonContent="Test name"
    id="configPanel__valueOptions"
    initialIsOpen={true}
    isLoading={false}
    isLoadingMessage={false}
    paddingSize="s"
  >
    <div
      className="euiAccordion euiAccordion-isOpen"
    >
      <div
        className="euiAccordion__triggerWrapper"
      >
        <button
          aria-controls="configPanel__valueOptions"
          aria-expanded={true}
          className="euiAccordion__button"
          id="random_html_id"
          onClick={[Function]}
          type="button"
        >
          <span
            className="euiAccordion__iconWrapper"
          >
            <EuiIcon
              className="euiAccordion__icon euiAccordion__icon-isOpen"
              size="m"
              type="arrowRight"
            >
              <EuiIconEmpty
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon euiAccordion__icon-isOpen"
                focusable="false"
                role="img"
                style={null}
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon euiAccordion__icon-isOpen"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </EuiIconEmpty>
            </EuiIcon>
          </span>
          <span
            className="euiIEFlexWrapFix"
          >
            Test name
          </span>
        </button>
      </div>
      <div
        aria-labelledby="random_html_id"
        className="euiAccordion__childWrapper"
        id="configPanel__valueOptions"
        role="region"
        tabIndex={-1}
      >
        <EuiResizeObserver
          onResize={[Function]}
        >
          <div>
            <div
              className="euiAccordion__padding--s"
            />
          </div>
        </EuiResizeObserver>
      </div>
    </div>
  </EuiAccordion>
</ConfigValueOptions>
`;
